compat/mingw.c

```diff
@@ -426,6 +426,43 @@ static wchar_t *normalize_ntpath(wchar_t *wbuf)
 	return wbuf;
 }
 
+/* rename and unlink a file. This is needed for the ability to recreate a file
+ * with the same name, for example on checkout which unlinks and recreates the file */
+static int rename_and_unlink(const wchar_t *pathname, int (*unlinker)(const wchar_t *))
+{
+	wchar_t temppath[MAX_LONG_PATH];
+	*temppath = 0;
+	/* leave some space for the .unlink_XXXXXX extension */
+	wcsncat(temppath, pathname, sizeof(temppath) - 15);
```

Comment:
For the sake of consistency, I suggest we use the same approach as in `is_dir_empty`. Something like this:

```
wchar_t wbuf[MAX_LONG_PATH + 15];
wcscpy(wbuf, wpath);
wcscat(wbuf, L".unlink_XXXXXX");
```

Reply:
But if we get a name that is too long, will MoveFile work?